example/spec/parallel/queueMaxExecutions/QueueMaxExecutionsSpec.js

```diff
@@ -0,0 +1,158 @@
+  let workflowExecution;
+
+  const providersDir = './data/providers/s3/';
+  const collectionsDir = './data/collections/s3_MOD09GQ_006';
+  const collection = { name: `MOD09GQ${testSuffix}`, version: '006' };
+  const provider = { id: `s3_provider${testSuffix}` };
+
+  const maxExecutions = 5;
+  const numberOfGranules = 6;
+
+  const lambdaStep = new LambdaStep();
+
+  process.env.CollectionsTable = `${config.stackName}-CollectionsTable`;
+  const collectionModel = new Collection();
+  process.env.ProvidersTable = `${config.stackName}-ProvidersTable`;
+  const providerModel = new Provider();
+
+  beforeAll(async () => {
+    const collectionJson = JSON.parse(fs.readFileSync(`${collectionsDir}/s3_MOD09GQ_006.json`, 'utf8'));
+    const granuleRegex = collectionJson.granuleId;
+    const collectionData = Object.assign({}, collectionJson, {
+      name: collection.name,
+      dataType: collectionJson.dataType + testSuffix,
+      provider_path: testDataFolder
+    });
+
+    const providerJson = JSON.parse(fs.readFileSync(`${providersDir}/s3_provider.json`, 'utf8'));
+    const providerData = Object.assign({}, providerJson, {
+      id: provider.id,
+      host: config.bucket
+    });
+
+    await Promise.all([
+      uploadGranulesData({
+        numberOfGranules,
+        granuleRegex,
+        bucket: config.bucket,
+        prefix: testDataFolder
+      }),
+      apiTestUtils.addCollectionApi({ prefix: config.stackName, collection: collectionData }),
+      apiTestUtils.addProviderApi({ prefix: config.stackName, provider: providerData })
+    ]);
+
+    workflowExecution = await buildAndExecuteWorkflow(
+      config.stackName,
+      config.bucket,
+      workflowName,
+      collection,
+      provider
+    );
+  });
+
+  afterAll(async () => {
+    await Promise.all([
+      collectionModel.delete(collection),
+      providerModel.delete(provider),
+      deleteFolder(config.bucket, testDataFolder)
+    ]);
+  });
+
+  it('executes successfully', () => {
+    expect(workflowExecution.status).toEqual('SUCCEEDED');
+  });
+
+  describe('respects the maximum amount of executions for the queue', () => {
+    let queueGranulesOutput;
+    let executionArns;
+
+    beforeAll(async () => {
+      queueGranulesOutput = await lambdaStep.getStepOutput(
+        workflowExecution.executionArn,
+        'QueueGranules'
+      );
+      executionArns = queueGranulesOutput.payload.running;
+    });
+
+    it('has expected count of queued executions', () => {
+      expect(executionArns.length).toEqual(numberOfGranules);
+    });
+
+    it('is only running the maximum number of executions', async () => {
+      // Wait until at least one of the executions has started
+      // TODO: can't rely on first execution ARN being started first
+      await waitForExecutionExists(executionArns[0], 5, 60);
+
+      // Get the state for all of the queued execution ARNs.
+      const getExecutionStates = executionArns
```

Comment:
Probably naive, but would it work to continuously query the executionExists like below and just make sure that the # of executions never tops the max?

Reply:
```
while ("RUNNING" in allExecutionStatus) {
  wait
  allExecutionStatus = ...
}